src/build_tools/hooks_hydra.py

```diff
@@ -64,6 +65,147 @@
 LOCAL_ARCH_SUFFIX = os.getenv('VSC_ARCH_SUFFIX')
 LOCAL_ARCH_FULL = f'{LOCAL_ARCH}{LOCAL_ARCH_SUFFIX}'
 
+VALID_TCGENS = ['2022a', '2023a']
+VALID_MODULES_SUBDIRS = VALID_TCGENS + ['system']
+VALID_TCS = ['foss', 'intel', 'gomkl', 'gimkl', 'gimpi']
+
+
+def get_tc_versions():
+    " build dict of (sub)toolchain-versions per valid generation "
+    tc_versions = {}
+    for toolcgen in VALID_TCGENS:
+        tc_versions[toolcgen] = []
+        for toolc in VALID_TCS:
+            try:
+                tc_versions[toolcgen].extend(get_toolchain_hierarchy({'name': toolc, 'version': toolcgen}))
+            except EasyBuildError:
+                # skip if no easyconfig found for toolchain-version
+                pass
+
+    return tc_versions
+
+
+def calc_tc_gen(name, version, tcname, tcversion, easyblock):
+    """
+    calculate the toolchain generation
+    return False if not valid
+    """
+    name_version = {'name': name, 'version': version}
+    toolchain = {'name': tcname, 'version': tcversion}
+    software = [name, version, tcname, tcversion, easyblock]
+
+    tc_versions = get_tc_versions()
+
+    # (software with) valid (sub)toolchain and version
+    for toolcgen in VALID_TCGENS:
+        if toolchain in tc_versions[toolcgen] or name_version in tc_versions[toolcgen]:
+            log_msg = f"Determined toolchain generation {toolcgen} for {software}"
+            return toolcgen, log_msg
+
+    # (software with) valid (sub)toolchain but invalid version
+    for toolcgen in VALID_TCGENS:
+        tcnames = [x['name'] for x in tc_versions[toolcgen]]
+        if toolchain['name'] in tcnames or name in tcnames:
+            log_msg = (f"Determined toolchain generation {toolcgen} for {software} is not valid."
+                       f" Choose one of {VALID_TCGENS}.")
+            return False, log_msg
```

Comment:
What is this check achieving? If the toolchain is not valid (not found in the previous `for` loop), can we just directly return `False` saying this toolchain version is not valid?

Reply:
the previous loop only checks for valid toolchain-version combinations, not for toolchains.
i'll update the comment to clarify this.

Reply:
loop removed in https://github.com/vub-hpc/build_tools/pull/31/commits/e358ee7c38b3c0b79213df258385ef3b2687dcf5